cpp/src/arrow/ipc/reader.h

```diff
@@ -301,9 +317,39 @@ class ARROW_EXPORT CollectListener : public Listener {
     return record_batches_;
   }
 
+  /// \return the all decoded metadatas
+  std::vector<std::shared_ptr<KeyValueMetadata>> metadatas() const { return metadatas_; }
```

Comment:
But why is this returning a copy? It could return a const-ref. Same for `record_batches()`...

Reply:
Ah, I just used the same signature as `record_batches()` but we should use const-ref.
I can't remember why I didn't use const-ref for `record_batches()` but I just doesn't notice it...

I also use const-ref for `record_batches()` too. It breaks a backward compatibility but I hope that nobody doesn't use this.